prime-router/src/test/kotlin/serializers/Hl7SerializerTests.kt

```diff
@@ -635,6 +636,19 @@ NTE|1|L|This is a final comment|RE"""
         }
     }
 
+    @Test
+    fun `test setTruncationLimitWithEncoding`() {
+
+        val testValueWithSpecialChars = "Test & Value ~ Text ^ String"
+        val testValueNoSpecialChars = "Test Value Text String"
+        val testLimit = 20
```

Comment:
Don't feel you have to define all the numbers separately.  next time you could do just
        assertEquals(serializer.getTruncationLimitWithEncoding(testValueWithSpecialChars, testLimit), 14)